lib/vsc/install/shared_setup.py

```diff
@@ -1497,19 +1497,19 @@ def parse_target(self, target, urltemplate=None):
             # soft pinning of (transitive) dependencies of prospector
             # ('~=' means stick to compatible release, https://www.python.org/dev/peps/pep-0440/#compatible-release);
             # updating these must be done in lockstep, see setup.cfg or pyproject.toml or whatever at:
-            # - https://github.com/PyCQA/pylint/blob/v2.10.2/setup.cfg
-            # - https://github.com/PyCQA/flake8/blob/3.8.4/setup.cfg
-            # - https://github.com/PyCQA/prospector/blob/1.5.0.1/pyproject.toml
+            # - https://github.com/PyCQA/pylint/blob/v2.12.2/setup.cfg
+            # - https://github.com/PyCQA/flake8/blob/3.9.2/setup.cfg
+            # - https://github.com/PyCQA/prospector/blob/1.5.3.1/pyproject.toml
             tests_requires.extend([
-                # stick to astroid < 2.8, as required by pylint 2.10.x
-                'astroid~=2.7.3',
-                # stick to pyflakes < 2.3.0, as required by prospector 1.5.0.x and flake8 3.8.x
-                'pyflakes~=2.2.0',
-                # stick to pycodestyle < 2.7.0, as required by flake8 3.8.x
-                'pycodestyle~=2.6.0',
-                'pylint~=2.10.2',
-                'flake8~=3.8.4',
-                'prospector~=1.5.0.1',
+                'pyflakes~=2.3.0',
+                'pycodestyle~=2.7.0',
+                'pylint~=2.12.2',
+                'flake8~=3.9.2',
+                'prospector',
+                'pylint-plugin-utils < 0.7',
```

Comment:
because there are version conflicts? the same reason we lock all the others? I found most of the old comments superfluous as the statement at the top says they need to be updated in lockstep.

Reply:
I understand why you removed the old comments, but when adding new restrictions like this it would be helpful to mention why they were added, so we know when they can be removed again?

Or were the comments that were there basically useless to you?

Reply:
the more sane option (well, the one I used) is to remove all dependencies except prospector and add them one by one as required until you get the new set.

Reply:
pylint-django is also new, you just did not notice :wink: . it get's pulled in but 2.5.0.1 conflicts with latest prospector, same with the plugins. astroid got remoted as it now just works.